newsroom/agenda/filters.py

```diff
@@ -0,0 +1,637 @@
+                name="workflow_status",
+            )
+        )
+        query.filter.append(
+            {
+                "bool": {
+                    "should": should_term_filters,
+                    "minimum_should_match": 1,
+                },
+            }
+        )
+    elif val == ["not intended"]:
+        query.filter.append(
+            nested_query(
+                path="coverages",
+                query={"terms": {"coverages.coverage_status": ["coverage not intended"]}},
+                name="coverage_status",
+            )
+        )
+
+
+def apply_agendas_filter(query: ESBoolQuery, val) -> None:
+    query.filter.append(
+        nested_query(
+            path="planning_items",
+            query={"terms": {get_aggregation_field("agendas"): val}},
+            name="agendas",
+        )
+    )
+
+
+def get_apply_agenda_filters(highlights: bool) -> SearchFilterFunction:
+    def _apply_agenda_filters(request: NewshubSearchRequest[AgendaSearchRequestArgs]) -> None:
+        if not request.args.filter:
+            return
+
+        query = request.search.post_filter if get_app_config("FILTER_BY_POST_FILTER", False) else request.search.query
+
+        for key, val in request.args.filter.items():
+            if not val:
+                continue
+            elif request.args.item_type == AgendaItemType.EVENT and key in planning_filters:
+                continue
+            elif key == "location":
+                apply_location_filter(query, val)
+            elif key == "coverage":
+                apply_coverage_filter(query, val)
+            elif key == "coverage_status":
+                apply_coverage_status_filter(query, val)
+            elif key == "agendas":
+                apply_agendas_filter(query, val)
+            else:
+                if request.args.item_type != AgendaItemType.EVENT:
+                    agg_field = get_aggregation_field(key)
+                    query.filter.append(
+                        {
+                            "bool": {
+                                "minimum_should_match": 1,
+                                "should": [
+                                    get_filter_query(key, val, agg_field, get_nested_config("agenda", key)),
+                                    nested_query(
+                                        path="planning_items",
+                                        query={"bool": {"filter": [{"terms": {f"planning_items.{agg_field}": val}}]}},
+                                        name=key,
+                                        inner_hits=highlights,
+                                    ),
+                                ],
+                            },
+                        }
+                    )
+                else:
+                    query.filter.append(
+                        get_filter_query(key, val, get_aggregation_field(key), get_nested_config("agenda", key))
+                    )
```

Comment:
A small suggestion here (not really relevant with functionality) would be to use a `match` / `case` statement since we're using python 3.10 :) What do you think? 

(Note: I didn't test it just changed it here)

```suggestion
            is_event_type = request.args.item_type == AgendaItemType.EVENT
            if not val or (is_event_type and key in planning_filters):
                continue
            
            match key:
                case "location":
                    apply_location_filter(query, val)
                case "coverage":
                    apply_coverage_filter(query, val)
                case "coverage_status":
                    apply_coverage_status_filter(query, val)
                case "agendas":
                    apply_agendas_filter(query, val)
                case _:
                    agg_field = get_aggregation_field(key)
                    if not is_event_type:
                        query.filter.append({
                            "bool": {
                                "minimum_should_match": 1,
                                "should": [
                                    get_filter_query(key, val, agg_field, get_nested_config("agenda", key)),
                                    nested_query(
                                        path="planning_items",
                                        query={"bool": {"filter": [{"terms": {f"planning_items.{agg_field}": val}}]}},
                                        name=key,
                                        inner_hits=highlights,
                                    ),
                                ],
                            },
                        })
                    else:
                        query.filter.append(
                            get_filter_query(key, val, agg_field, get_nested_config("agenda", key))
                        )

```